src/generator/AutoRest.Go/Model/ParameterGo.cs

```diff
@@ -186,9 +186,30 @@ public static IEnumerable<ParameterGo> HeaderParameters(this IEnumerable<Paramet
             return parameters.ByLocationAsRequired(ParameterLocation.Header, isRequired);
         }
 
+         public static IEnumerable<ParameterGo> URLParameters(this IEnumerable<ParameterGo> parameters)
+        {
+            var urlParams = new List<ParameterGo>();
+            foreach (ParameterGo p in parameters.ByLocation(ParameterLocation.Path))
+            {
+                if (p.Method.CodeModel.BaseUrl.Contains(p.SerializedName))
+                {
+                    urlParams.Add(p);
```

Comment:
[syntax sugar opportunity]
Instead of allocating `urlParams` you could rewrite this function to look like:

``` C#
public static IEnumerable<ParameterGo> URLParameters(this IEnumerable<ParameterGo> parameters)
{
    foreach (ParameterGo p in parameters.ByLocation(ParameterLocation.Path))
    {
        if (p.Method.CodeModel.BaseUrl.Contains(p.SerializedName))
        {
            yield return p;
        }
    }
}
```

Alternatively, you could embrace LINQ:

``` C#
public static IEnumerable<ParameterGo> URLParameters(this IEnumerable<ParameterGo> parameters)
{
    return parameters.ByLocation(ParameterLocation.Path).Where(p => p.Method.CodeModel.BaseUrl.Contains(p.SerializedName));
}
```

Needless to say, these suggestions aren't even nits, just showing off cool things about C# :)